apps/web/src/components/SearchRequestTable/components/SearchRequestsTableFilter.test.tsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * @jest-environment jsdom
+ */
+import "@testing-library/jest-dom";
+
+import React from "react";
+import { Provider as GraphqlProvider } from "urql";
+import { fromValue } from "wonka";
+
+import { axeTest, renderWithProviders } from "@gc-digital-talent/jest-helpers";
+import { act, screen } from "@testing-library/react";
+import {
+  fakeClassifications,
+  fakeDepartments,
+} from "@gc-digital-talent/fake-data";
+import { PoolStream } from "@gc-digital-talent/graphql";
+
+import { selectFilterOption, submitFilters } from "~/utils/jestUtils";
+
+import { SearchRequestsTableFilterDialog } from "./SearchRequestsTableFilterDialog";
+
+const departments = [fakeDepartments()[0]];
+const classifications = [fakeClassifications()[0]];
+
+const mockSubmit = jest.fn();
+const mockToggle = jest.fn();
+const mockClient = {
+  executeQuery: () =>
+    fromValue({
+      data: {
+        departments,
+        classifications,
+      },
+    }),
+  // See: https://github.com/FormidableLabs/urql/discussions/2057#discussioncomment-1568874
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+} as any;
+
+const renderSearchRequestsTableFilterDialog = () =>
+  renderWithProviders(
+    <GraphqlProvider value={mockClient}>
+      <SearchRequestsTableFilterDialog
+        onSubmit={mockSubmit}
+        onOpenChange={mockToggle}
+        isOpen
+        activeFilters={{
+          departments: [],
+          status: [],
+          classifications: [],
+          streams: [],
+        }}
+      />
+    </GraphqlProvider>,
+  );
+
+describe("SearchRequestsTableFilterDialog", () => {
+  it("should have no accessibility errors", async () => {
+    await act(async () => {
+      const { container } = renderSearchRequestsTableFilterDialog();
+      await axeTest(container);
+    });
+  });
+
+  it("Should render the filter", async () => {
+    await act(async () => {
+      renderSearchRequestsTableFilterDialog();
+    });
+
+    // assert filter has everything present
+    expect(screen.getAllByText("Status")).toBeTruthy();
+    expect(screen.getAllByText("Departments")).toBeTruthy();
+    expect(screen.getAllByText("Classifications")).toBeTruthy();
+    expect(screen.getAllByText("Streams")).toBeTruthy();
+    expect(screen.getAllByText("Clear filters")).toBeTruthy();
+    expect(screen.getAllByText("Show results")).toBeTruthy();
+    expect(screen.getAllByRole("combobox")).toHaveLength(4);
```

Comment:
We can bake in some accessibility checks with querying elements by role and accessible name. This way be confirm the role as well as the accessible name instead of the rendered text.

```tsx
expect(await screen.getbyRole("combox", { name: /status/i })).toBeInTheDocument();

expect(await screen.getbyRole("button", { name: /clear filters/i })).toBeInTheDocument();
```

Reply:
Aye aye sounds good, done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/6897/commits/408b77e03335fc72bbb0dac34edb8a3a80c917da